### PR TITLE
Add cpp file parameter to utilities for creating CppDoc members

### PR DESCRIPTION
**Pull request by tiffany1618**
- Update CppWriter utilities to account for changes in #295
- Fix the name of the output file in `CppWriter.writeCppFile`

### CODE REVIEW

**Review by bocchino — APPROVED** [review]
Looks good!

**Comment by tiffany1618** [comment]
@bocchino I think we should also append a file extension to selected cppFile names, to make it consistent with the default cpp file names that are passed to a CppDoc.

File: compiler/lib/src/main/scala/codegen/CppWriter/CppWriter.scala [hunk 1]
```diff
@@ -61,7 +61,11 @@ object CppWriter extends LineUtils{
     cppFile: Option[String] = None
   ) = {
     val lines = CppDocCppWriter.visitCppDoc(cppDoc, cppFile)
-    writeLinesToFile(s, cppFile.getOrElse(cppDoc.cppFileName), lines)
```
[the remark this answers is not in the document]
Comment (bocchino) [reply compiler/lib/src/main/scala/codegen/CppWriter/CppWriter.scala]: Alternatively we could keep this change and rename the parameter `cppFile` to `cppFileNameBase`. We could also rename `fileName` to `fileNameBase` in `createCppDoc`.

[the remark this answers is not in the document]
Comment (tiffany1618) [reply compiler/lib/src/main/scala/codegen/CppWriter/CppWriter.scala]: Ok, this sounds good! I've renamed the parameters.

**Review by bocchino — APPROVED** [review]
Looks good!

**Pull request merged**